src/file/path.rs

```diff
@@ -0,0 +1,160 @@
+        separator: String,
+        /// Arguments expression is a sequence of argument tokens.
+        ///
+        /// See [`ArgumentTokenSequence`] for details.
+        arguments_expression: ArgumentTokenSequence,
+    },
+    /// Comment is a string starting with hash (`#`) symbol.
+    ///
+    /// Please, note that genuine understanding of comment lines in [ssh_config(5)](https://linux.die.net/man/5/ssh_config)
+    /// is different from what we call a comment, because we distinguish [`Empty`] expression as a separate case, not as a comment.
+    Comment(String),
+    /// Empty expression is an empty string.
+    Empty,
+    /// Any string not being a valid [`ConfigurationOptions`], [`Comment`] or [`Empty`] expression.
+    Malformed(String),
+}
+
+impl fmt::Display for Expression {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::ConfigurationOptions {
+                keyword,
+                separator,
+                arguments_expression,
+            } => write!(
+                f,
+                "{}{}{}",
+                keyword,
+                separator,
+                arguments_expression
+                    .iter()
+                    .map(|c| c.to_string())
+                    .collect::<Vec::<String>>()
+                    .join("")
+            ),
+            Expression::Comment(comment) => write!(f, "{}", comment),
+            Expression::Empty => write!(f, ""),
+            Expression::Malformed(malformed) => write!(f, "{}", malformed),
+        }
+    }
+}
+
+/// Arguments expression is a sequence of argument tokens.
+///
+/// It must contain at least one non-[`Whitespace`] token.
+/// It may not contain several [`Pure`] / [`Quoted`] or [`Whitespace`] tokens in a row.
+type ArgumentTokenSequence = Vec<ArgumentToken>;
+
+/// Argument token is a part of arguments expression string.
+///
+/// More specifically, it is either an argument, or a whitespace sequence splitting one argument from another.
+#[derive(PartialEq, Debug, Clone)]
+pub enum ArgumentToken {
+    /// Pure token is a value that may not contain whitespace characters and quoting characters.
+    Pure(String),
+    /// Quoted token is a value quoted with double-quote (`"`) symbol.
+    ///
+    /// It may contain whitespace characters inside, but double-quote characters must be escaped (`\"`).
+    Quoted(String),
+    /// Whitespace token is a separator for [`Pure`] and [`Quoted`] tokens.
+    Whitespace(WhitespaceString),
+}
+
+impl fmt::Display for ArgumentToken {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ArgumentToken::Pure(value) => write!(f, "{}", value),
+            ArgumentToken::Quoted(value) => write!(f, "\"{}\"", value),
+            ArgumentToken::Whitespace(value) => write!(f, "{}", value),
+        }
+    }
+}
+
+/// Whitespace string is a string consisting only of space (`' '`) or tabular (`'\t'`) symbols
```

Comment:
type alias is good, but it would be better to have strict validation of this rule. let's consider making a type wrapper, so it would  panic